src/libraries/System.Text.Json/tests/Common/RequiredKeywordTests.cs

```diff
@@ -664,6 +664,53 @@ public class ClassWithCustomRequiredPropertyName
             public required int PropertyWithInitOnlySetter { get; init; }
         }
 
+        [Fact]
+        public async Task DerivedClassWithRequiredProperty()
+        {
+            var options = Serializer.CreateOptions(includeFields: true);
+            var value = new DerivedClassWithRequiredInitOnlyProperty { MyInt = 42, MyBool = true, MyString = "42", MyProp = 42.0M, MyLong = 4242, MyMember1 = 1, MyMember2 = 2, MyField = "42" };
+            string json = await Serializer.SerializeWrapper(value, options);
+            Assert.Equal("""{"MyInt":42,"MyBool":true,"MyString":"42","MyProp":42.0,"MyLong":4242,"MyMember1":1,"MyMember2":2,"MyField":"42"}""", json);
+
+            value = await Serializer.DeserializeWrapper<DerivedClassWithRequiredInitOnlyProperty>(json, options);
+            Assert.Equal(42, value.MyInt);
+            Assert.True(value.MyBool);
+            Assert.Equal("42", value.MyString);
+            Assert.Equal(42.0M, value.MyProp);
+            Assert.Equal(4242, value.MyLong);
+            Assert.Equal(1, value.MyMember1);
+            Assert.Equal(2, value.MyMember2);
+            Assert.Equal("42", value.MyField);
+        }
+
+        public class BaseClassWithInitOnlyProperty
+        {
+            public int MyInt { get; init; }
+            public bool MyBool { get; init; }
+            public string MyString { get; set; }
+            public string MyProp { get; init; }
+
+            public string MyMember1;
+            public string MyMember2 { get; init; }
+
+            public string MyField;
```

Comment:
The base member must be defined as a member initializer to trigger the bug condition
```suggestion
            public required int MyField;
```